src/systemcmds/tests/tests_main.c

```diff
@@ -262,13 +262,27 @@ int tests_main(int argc, char *argv[])
 		return 1;
 	}
 
+	size_t tone_test_index = 0;
+	char *tone_test = {"tone"};
+	char *tone_fail[2] = {NULL, "2"};
+	char *tone_pass[2] = {NULL, "14"};
+
+	// Identify the tone test index for later use.
+	for (size_t i = 0; tests[i].name; i++) {
+		if (*tone_test == *tests[i].name) {
+			tone_test_index = i;
+		}
+	}
+
 	for (unsigned i = 0; tests[i].name; i++) {
 		if (!strcmp(tests[i].name, argv[1])) {
 			if (tests[i].fn(argc - 1, argv + 1) == 0) {
+				tests[tone_test_index].fn(2, tone_pass); // Play a notification.
```

Comment:
Is the tone guaranteed to be linked?
If not we need a guard on `int tone_test_index == -1`  So if tone is not found to not invoke the test at 0.

Reply:
Nice catch. Thank you!  Guards added, initialized to -1.

I made another pass through test_runner  and changed loop index types to size_t (index type) and condensed a few printf() calls as well.  Those changes are pushed up and rebased with current master.  Tested on pixhawk 4 (V5) and performing as expected - **"tests time"** still fails to return properly as reported in #11361, (not caused by this PR).